packages/apollo-server-express/src/ApolloServer.ts

```diff
@@ -142,12 +161,40 @@ export class ApolloServer extends ApolloServerBase {
     // schema, you'll need to manually specify `introspection: true` in the
     // ApolloServer constructor; by default, the introspection query is only
     // enabled in dev.
-    const guiEnabled =
-      !!gui || (gui === undefined && process.env.NODE_ENV !== 'production');
-
     app.use(path, (req, res, next) => {
-      if (guiEnabled && req.method === 'GET') {
+      let partialGuiOverrides: Partial<GuiOptions>;
+      if (!gui) {
+        partialGuiOverrides = {};
+      } else if (isPartialGui(gui)) {
+        partialGuiOverrides = gui;
+      } else {
+        partialGuiOverrides = gui(req);
+      }
+
+      console.log('partial enabled', partialGuiOverrides.enabled);
```

Comment:
Let's remove the console.log's before we merge

Reply:
Whoops! Thanks for catching.